tests/test_xtuml/test_metamodel.py

```diff
@@ -66,6 +68,23 @@ def test_select_any_where(self):
         m = self.metamodel
         inst = m.select_any('S_DT', where(Name='void'))
         self.assertEqual(inst.Name, 'void')
+
+    def test_select_many_ordered_by(self):
+        m = self.metamodel
+        q = m.select_many('S_DT', None, order_by('Name', 'DT_ID'))
```

Comment:
I was thinking, it should be possible to to make the where_clause and order_by argument less independent on the order in the parameter list. Something like this:

def select_many(kind, set_modifiers*):
   s = map(apply_set_operator, set_modifiers)
   return s

then you could type:
    q = m.select_many('S_DT', order_by('Name', 'DT_ID'), where(...))
or
    q = m.select_many('S_DT', where(...), order_by('Name', 'DT_ID'))
or
    q = m.select_many('S_DT', order_by('Name', 'DT_ID'))

Reply:
hmmm. I will give this some thought. Of course the difficulty is how to implement the 'apply_set_operator' part

Reply:
map() might have been a bad idea. I slapped together a gist demonstrating one way of doing it. needs some cleaning though...

See https://gist.github.com/john-tornblom/21eca04788c3980778039bf91866e7dd